src/main/java/de/tum/in/www1/artemis/service/ComplaintService.java

```diff
@@ -53,7 +53,7 @@ public Complaint createComplaint(Complaint complaint, Principal principal) {
         Result originalResult = resultRepository.findByIdWithEagerFeedbacksAndAssessor(complaint.getResult().getId())
                 .orElseThrow(() -> new BadRequestAlertException("The result you are referring to does not exist", ENTITY_NAME, "resultnotfound"));
         StudentParticipation studentParticipation = (StudentParticipation) originalResult.getParticipation();
-        User student = studentParticipation.getStudent();
+        User student = studentParticipation.getStudent().get(); // TODO: adapt for teams
```

Comment:
Good question. I think adapting the complaints to teams is probably better suited to a follow-up PR since it might require some non-trivial decisions on how we want to handle complaints for team-based exercises. Also this PR is already on the larger side and since the creation of team-based exercises is still hidden for non-admin users, it is not like regular users will run into this under-construction part of the epic. I should still see about though that I deliver this within some days.

Reply:
Alright, then let's handle this in a different PR.